clients/spark-extensions-basetests/src/main/java/org/projectnessie/spark/extensions/AbstractNessieSparkSqlExtensionTest.java

```diff
@@ -573,4 +576,90 @@ void testValidCatalog() {
       spark.sessionState().catalogManager().setCurrentCatalog(catalog);
     }
   }
+
+  @ParameterizedTest
+  @CsvSource({
+    "testCompaction,tbl",
+    "main,tbl",
+    "testCompaction,`tbl@testCompaction`",
+    "main,`tbl@main`"
+  })
+  void testCompaction(String branchName, String tableName) throws NessieNotFoundException {
+    String branchHash = prepareForCompaction(branchName);
+
+    if (spark.version().compareTo("3.2.0") < 0) {
```

Comment:
The ugly Spark 3.1 behavior should go into a `ITNessieStatements` for Spark 3.1 and not pollute this test IMO.

Reply:
Agree. Handled.